src/Chapter1.hs

```diff
@@ -359,301 +277,250 @@ scripts together.
   expressions. The syntax of guards is the following:
-
 @
 sign :: Int -> Int
 sign n
     | n < 0 = (-1)
     | n == 0 = 0
     | otherwise = 1
 @
-
 You define different conditions in different branches, started by the '|'
 symbol. And the functions check them from top to bottom, returning the first
 value after "=" where the condition is true.
-
 ♫ NOTE: The "=" sign goes after each branch, respectively.
-
 ♫ NOTE: It is essential to have the same indentation before each branch "|"!
   Remember, that Haskell is indentation- and layout-sensitive language.
-
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+  | x > y = mid y x z  -- mid 3 1 2 = mid 1 3 2 --
+  | y > z = mid x z y  -- mid 1 3 2 = mid 1 2 3 --
+  | otherwise = y      -- mid 1 3 2 = 2 --
 
 {- |
 =⚔️= Task 8
-
 Implement a function that checks whether a given character is a vowel.
-
 🕯 HINT: use guards
-
 >>> isVowel 'a'
 True
 >>> isVowel 'x'
 False
 -}
-isVowel c = error "isVowel: not implemented!"
-
+isVowel :: Char -> Bool
+isVowel c
+  | c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' = True
+  | otherwise = False
 
 {- |
 == Local variables and functions
-
 So far, we've been playing only with simple expressions and function
 definitions. However, in some cases, expressions may become complicated, and it
 could make sense to introduce some helper variables.
-
 You can use the let-in construction in Haskell to define variables.
 Here goes an example:
-
 @
 half :: Int -> Int
 half n = let halfN = div n 2 in halfN
 @
-
 ♫ NOTE: __let-in__ is also an expression! You can't just define variables; you
   also need to return some expression that may use defined variables.
-
 The syntax for defining multiple variables requires to care about indentation
 more, but there is nothing special in it as well:
-
 @
 halfAndTwice :: Int -> (Int, Int)
 halfAndTwice n =
     let halfN = div n 2
         twiceN = n * 2
     in (halfN, twiceN)
 @
-
 In addition to let-in (or sometimes even alternatively to let-in) you can use
 the __where__ construction to define local variables and functions.
 And, again, the example:
-
 @
 pythagoras :: Double -> Double -> Double
 pythagoras a b = square a + square b
   where
     square :: Double -> Double
     square x = x ^ 2
 @
-
 You can define multiple functions inside __where__!
 Just remember to keep proper indentation.
 -}
 
 {- |
 =⚔️= Task 9
-
 Implement a function that returns the sum of the last two digits of a number.
-
 >>> sumLast2 42
 6
 >>> sumLast2 134
 7
 >>> sumLast2 1
 1
-
 Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = lastDigit + secLastDigit
+  where
+    lastDigit = mod n 10
+    secLastDigit = div (mod n 100 - lastDigit) 10
```

Comment:
Great!
For `div` the last digit won't matter, so it is okay not to subtract it in here 🙂